### ISSUE TITLE
[sklearn#34639] CI PyGitHub DeprecationWarning in `maint_tools/update_tracking_issue.py`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#305]
**Mirrored from [scikit-learn#34639](https://github.com/scikit-learn/scikit-learn/issues/34639)**

Original title: CI PyGitHub DeprecationWarning in `maint_tools/update_tracking_issue.py`
Original labels: Needs Triage
Created: 2026-08-03T08:21:12Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

Noticed this in one CI build log:
```
/home/runner/work/scikit-learn/scikit-learn/maint_tools/update_tracking_issue.py:68: DeprecationWarning: Argument login_or_token is deprecated, please use auth=github.Auth.Token(...) instead   gh = Github(args.bot_github_token)
```

Other scripts using PyGithub uses it the right way:
```py
gh = Github(auth=Auth.Token(token))
```

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Addressed in upstream PR(s). See scikit-learn/scikit-learn PRs for #34639.
